src/main/java/org/opensearch/ad/transport/ADStatsNodesTransportAction.java

```diff
@@ -126,6 +109,29 @@ private ADStatsNodeResponse createADStatsNodeResponse(ADStatsRequest adStatsRequ
             }
         }
 
-        return new ADStatsNodeResponse(clusterService.localNode(), statValues);
+        return new ADStatsNodeResponse(sdkClusterService.localNode(), statValues);
+    }
+
+    @Override
+    protected void doExecute(Task task, ADStatsNodeRequest request, ActionListener<ADStatsNodesResponse> actionListener) {
+        ActionListener<ADStatsNodesResponse> listener = wrapRestActionListener(actionListener, FAIL_TO_GET_NODE_STATS);
+        ADStatsNodeResponse adStatsNodeResponse = null;
+        List<FailedNodeException> failures = new ArrayList<>();
+        try {
+            adStatsNodeResponse = nodeOperation(request);
+        } catch (Exception e) {
+            if (e instanceof FailedNodeException) {
+                failures.add((FailedNodeException) e);
+                logger.info("Failure in get Node stats due to exception", e);
+            } else {
+                logger.info("Could not process get Node stats API due to exception", e);
+            }
+            listener.onFailure(e);
+        }
+        List<ADStatsNodeResponse> responses = new ArrayList<>();
+        if (adStatsNodeResponse != null) {
+            responses.add(adStatsNodeResponse);
+        }
+        listener.onResponse(newResponse(responses, failures));
```

Comment:
Based on [this](https://github.com/opensearch-project/anomaly-detection/pull/857/files#r1160094000) we might won't need `doExecute` here.

Reply:
As of now, we need doExecute method as We have to cater all the aspects needed to implement the above referenced work.

Reply:
Let's add a TODO and link the issue here.

Reply:
https://github.com/opensearch-project/opensearch-sdk-java/issues/683

Reply:
Done added in PR as well